crypto/cmp/cmp_ctx.c

```diff
@@ -941,6 +941,10 @@ int OSSL_CMP_CTX_set_option(OSSL_CMP_CTX *ctx, int opt, int val)
         ctx->unprotectedErrors = val;
         break;
     case OSSL_CMP_OPT_VALIDITY_DAYS:
+        if (val >= (INT_MAX - time(NULL)) / (60 * 60 * 24)) {
```

Comment:
INT_MAX for time seems like a year 2038 problem

Reply:
Correct. I thought it should be solved on a more general basis within all of OpenSSL.
Or has this already been done for any/all similar such cases, and if so, how?

Reply:
I'm not aware of either any other problems or fixes, but it's probably not a good idea to add more problems. As far as I know, there is no portable way in C to check for this. time_t doesn't even need to be an integer, but it is on all platforms we support. On some platforms it's unsigned.

Reply:
Other parts of the code probably use an ASN1_TIME

Reply:
I've fixed the year 2038 issue by using `ASN1_TIME_adj()` as described in the first example in `doc/man3/ASN1_TIME_set.pod`.